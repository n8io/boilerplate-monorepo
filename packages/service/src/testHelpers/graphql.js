import { PubSub } from 'apollo-server-express';
import { createTestClient } from 'apollo-server-testing';
import { subscribe as graphqlSubscribe } from 'graphql';
import { has, head, keys, path, pipe, prop, __ } from 'ramda';
import { makeServer } from '../server';
import { make as makeContext } from './context';

const makeGraphqlClient = (context) => {
  const server = makeServer({ context });
  const { query, mutate } = createTestClient(server);

  const subscribe = async (subscription, pubsub) => {
    const iterator = await graphqlSubscribe(
      server.schema,
      subscription,
      {},
      { ...context, pubsub }
    );

    if (has('errors', iterator)) return iterator;

    const { value } = await iterator.next();

    return value;
  };

  const whileSubscribed = (subscription, callback) => {
    const pubsub = new PubSub();
    const promise = subscribe(subscription, pubsub);

    callback(pubsub);
    return promise;
  };

  return {
    context,
    execMutation: mutate,
    execQuery: query,
    whileSubscribed,
  };
};

const graphqlClient = makeGraphqlClient(makeContext());

const responseToData = (response) => {
  // eslint-disable-next-line no-console
  response.errors && console.error(response.errors);
  expect(response.errors).toBeUndefined();
  expect(response.data).not.toBeNull();

  return pipe(keys, head, prop(__, response.data))(response.data);
};

const responseToErrorCode = path(['errors', 0, 'extensions', 'code']);

export {
  graphqlClient,
  makeGraphqlClient,
  responseToData,
  responseToErrorCode,
};
