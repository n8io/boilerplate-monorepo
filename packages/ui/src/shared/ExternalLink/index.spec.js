import React from 'react';
import { render } from 'testHelpers';
import { ExternalLink } from '.';

describe('<ExternalLink/>', () => {
  const children = 'TEXT';

  const defaultProps = {
    children,
    href: 'HREF',
  };

  const renderComponent = (overrides) =>
    render(<ExternalLink {...defaultProps} {...overrides} />);

  test('renders children', () => {
    const { container } = renderComponent();

    expect(container.firstChild).toMatchSnapshot();
  });
});
