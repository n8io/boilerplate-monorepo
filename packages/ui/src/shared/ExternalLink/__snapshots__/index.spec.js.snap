// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExternalLink/> renders children 1`] = `
<a
  class="ExternalLink__StyledLink-dCpmsY cMdWQu"
  href="HREF"
  rel="noopener noreferrer"
  target="_external"
>
  TEXT
</a>
`;
